src/evo/deterministicmns.cpp

```diff
@@ -332,8 +331,8 @@ void CDeterministicMNList::PoSePunish(const uint256& proTxHash, int penalty, boo
                   __func__, proTxHash.ToString(), dmn->pdmnState->nPoSePenalty, newState->nPoSePenalty, maxPenalty);
     }
 
-    if (newState->nPoSePenalty >= maxPenalty && newState->nPoSeBanHeight == -1) {
-        newState->nPoSeBanHeight = nHeight;
+    if (newState->nPoSePenalty >= maxPenalty && !newState->IsBanned()) {
+        newState->BanIfNotBanned(nHeight);
         if (debugLogs) {
```

Comment:
Since that was a little redundant I initially changed `BanIfNotBanned` to return true if it banned, and false if it didn't ban and then changed this to below. But when I thought about it more, I wasn't sure it would be worth returning something from `BanIfNotBanned` so kept it how it is currently even if it is a little redundant maybe
```suggestion
    if (newState->nPoSePenalty >= maxPenalty) {
        if (newState->BanIfNotBanned(nHeight) && debugLogs) {
```